ldk/javascript/examples/self-test-loop/src/tests/index.ts

```diff
@@ -814,28 +827,108 @@ export const networkHTTP = (): Promise<boolean> =>
       });
   });
 
+const finaliseWebsocketTest = async (cancellable: Cancellable, socket: network.Socket) => {
+  try {
+    cancellable.cancel();
+    await socket.close((error) => {
+      console.error(`Received error while closing websocket: ${error.message}`);
+    });
+  } catch (e) {
+    console.error(`Received error while finalising websocket: ${e.message}`);
+  }
+};
+
+export const networkWebSocket = (): Promise<boolean> =>
+  new Promise(async (resolve, reject) => {
+    const url = 'wss://html5rocks.websocket.org/echo';
+    const testData = new Uint8Array([53, 6, 6, 65, 20, 74, 65, 78, 74]);
+    const testText = 'some text';
+    let textTestPassed = false;
+    let binaryTestPassed = false;
+
+    setTimeout(() => {
+      reject(new Error('Network websocket test did not finish in the appropriate timespan.'));
+    }, 20000);
+
+    const socketConfiguration: network.SocketConfiguration = {
+      url: url,
+      useCompression: true,
+    };
+
+    try {
+      const socket = await network.webSocketConnect(socketConfiguration);
```

Comment:
How did you get around our linter settings with mixing `async`/`await` and `.then()`?